app/src/controllers/accessRequest.ts

```diff
@@ -1,29 +1,48 @@
+import { AccessRole } from '../utils/enums/application';
 import { userService, accessRequestService } from '../services';
 
-import type { NextFunction, Request, Response } from 'express';
-
-import type { UserAccessRequest } from '../types';
+import type { NextFunction, Request, Response } from '../interfaces/IExpress';
+import type { JwtPayload } from 'jsonwebtoken';
+import type { AccessRequest, User, UserAccessRequest } from '../types';
 
 const controller = {
   // Request to create user & access
-  createUserAccessRevokeRequest: async (req: Request, res: Response, next: NextFunction) => {
-    // TODO check if the calling user is a supervisor or an admin
+  createUserAccessRevokeRequest: async (
+    req: Request<never, never, { user: User; accessRequest: AccessRequest }>,
+    res: Response,
+    next: NextFunction
+  ) => {
+    // Check if the requestee is an admin
+    const admin =
+      (req.currentUser?.tokenPayload as JwtPayload)?.client_roles?.some(
+        (role: string) => role === AccessRole.PCNS_DEVELOPER || role === AccessRole.PCNS_ADMIN
+      ) ?? false;
+
     try {
       let response;
       const { user, accessRequest } = req.body;
       if (accessRequest?.grant === false) {
         response = await accessRequestService.createUserAccessRevokeRequest(accessRequest);
```

Comment:
May need a check to prevent duplicate requests to `createUserAccessRevokeRequest`. ie. two revoke/grant requests coming in for the same person at different times. Need to to the same for `createUserAccessRevokeRequest`